modules/custom/movie_reservation/src/Controller/MovieReservationController.php

```diff
@@ -1,20 +1,32 @@
 <?php
-
+ 
 namespace Drupal\movie_reservation\Controller;
-
-class MovieReservationController{
-
+use \Drupal\node\Entity\Node;
+use Symfony\Component\HttpFoundation\Request;
+  
+  class MovieReservationController {
+  
     public function movieReservation() {
-   
+  
+      $genres = \Drupal::entityTypeManager()
+        ->getStorage('taxonomy_term')
+        ->loadByProperties(['vid' => 'genres']);
+      $genre_id = \Drupal::request()->query->get('genre');
+
       $query = \Drupal::entityQuery('node')
-        ->condition('type','movie');
-      $movies = \Drupal\node\Entity\Node::loadMultiple($query->execute());
+        ->condition('type', 'movie');
+      
+      if(isset($genre_id)){
+        $query->condition('field_genre', $genre_id);
+      }
```

Comment:
Sta predstavlja ovaj deo koda?

Reply:
Posto ste mi poslali da popravim debugger, skontao sam da mi id vuce nekad null. A isset je funkcija koja mi proverava da li je id postavaljena, sto znaci da mora biti deklarisana i da nije null. Vraca true ako id postoji i nije null, u suprotnom vraca false. A postavio sam i query koji ce da mi filtirira i kupi podatke(id) iz odredjeno polja, field_genre.

Reply:
U kom slucaju ce ti ID biti null, a u kom nece?

Reply:
Bio mi je id null jer sam skontao da ne gadjam dobro polje,tj kad sam dodao genre u content type, genrisano machine name mi je bilo field_genre dok sam ja trazio field_data, kada sam to ispravio nisam dobio id za null.